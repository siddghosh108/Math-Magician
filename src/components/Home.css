html {
  background-color: rgb(255, 251, 245);
  color: #968893;
  height: 200px;
}

h2 {
  font-size: 38px;
  text-align: center;
  padding-top: 3rem;
  margin-bottom: 3rem;
  color: #7c707a;
}

.text-container p {
  text-align: center;
  font-size: 24px;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  margin: 50px;
}

.math {
  color: #867e85;
  padding-top: 5rem;
  font-size: 38px;
  text-align: center;
}
