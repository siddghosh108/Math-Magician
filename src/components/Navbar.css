nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 15px 20px;
  background-color: rgb(245, 232, 234);
}

nav h1 {
  color: #75085e91;
  font-size: 40px;
  margin-left: 20px;
}

nav ul {
  display: flex;
}

nav ul li {
  list-style: none;
  margin: 0 20px;
  font-size: 30px;
}

nav ul li a {
  text-decoration: none;
  color: #574e5591;
}

nav ul li:hover {
  text-decoration: underline;
  color: #ff07d1;
}
