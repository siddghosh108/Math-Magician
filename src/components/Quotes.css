.quotes-container {
  text-align: center;
  margin: 15rem;
}

.quote,
.author {
  font-size: 30px;
  color: #6e646c;
}

.loading {
  text-align: center;
  font-size: 32px;
  color: #ff07d1;
  margin: 5rem;
}

.author {
  padding-top: 1rem;
}
