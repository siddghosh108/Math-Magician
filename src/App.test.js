import {
  render, screen, waitFor, act,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import Items from './components/calcItems';
import Calculator from './components/Calculator';
import CalculatorNew from './components/CalculatorNew';
import Home from './components/Home';
import Navbar from './components/Navbar';
import QuotePage from './components/Quote';
import Quotes from './components/Quotes';
import calculate from './logic/calculate';
import operate from './logic/operate';

describe('tests math-magician App test', () => {
  test('renders the Items component', () => {
    const handleClick = jest.fn();
    const number = 8;

    const componentTree = renderer.create(
      <Items number={number} handleClick={handleClick} />,
    );

    expect(componentTree.toJSON()).toMatchSnapshot();
  });

  test('should render the component', () => {
    const component = renderer.create(<Calculator />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render the navbar and calculator page correctly', () => {
    render(
      <BrowserRouter>
        <CalculatorNew />
      </BrowserRouter>,
    );
    const navbar = screen.getByText('Math Magicians');
    expect(navbar).toBeInTheDocument();
    const calculator = screen.getByRole('main');
    expect(calculator).toBeInTheDocument();
  });

  test('should render the welcome message', () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>,
    );

    const welcomeMessage = screen.getByText(/Welcome to Math World!/i);
    expect(welcomeMessage).toBeInTheDocument();

    expect(welcomeMessage).toMatchSnapshot();
  });

  test('should render the paragraph text', () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>,
    );

    const paragraphText = screen.getByText(/Lorem ipsum dolor sit amet/i);
    expect(paragraphText).toBeInTheDocument();

    expect(paragraphText).toMatchSnapshot();
  });

  test('renders the Navbar component', () => {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );

    const headerText = screen.getByText(/Math Magicians/i);
    expect(headerText).toBeInTheDocument();

    const homeLink = screen.getByRole('link', { name: /Home/i });
    expect(homeLink).toBeInTheDocument();

    const calculatorLink = screen.getByRole('link', { name: /Calculator/i });
    expect(calculatorLink).toBeInTheDocument();

    const quoteLink = screen.getByRole('link', { name: /Quote/i });
    expect(quoteLink).toBeInTheDocument();

    const navbarComponent = screen.getByRole('navigation');
    expect(navbarComponent).toMatchSnapshot();
  });

  test('renders the QuotePage component', () => {
    render(
      <MemoryRouter>
        <QuotePage />
      </MemoryRouter>,
    );

    const navbarElement = screen.getByRole('navigation');
    expect(navbarElement).toBeInTheDocument();

    const quotesElement = screen.getByRole('main');
    expect(quotesElement).toBeInTheDocument();

    const componentTree = renderer.create(
      <MemoryRouter>
        <QuotePage />
      </MemoryRouter>,
    );
    expect(componentTree.toJSON()).toMatchSnapshot();
  });

  const waitForFetch = (data) => {
    global.fetch = jest.fn(() => Promise.resolve({
      json: () => Promise.resolve(data),
    }));
  };

  test('shows charging status while getting data', async () => {
    waitForFetch([]);

    render(<Quotes />);

    const loadingElement = await screen.findByText(/Quote is loading/i);
    expect(loadingElement).toBeInTheDocument();
  });

  test('show citations and authors after getting the data', async () => {
    waitForFetch([
      {
        id: 1,
        quote: 'Cita de ejemplo',
        author: 'Autor de ejemplo',
      },
    ]);

    await act(async () => {
      render(<Quotes />);
    });

    await waitFor(() => {
      const quoteElement = screen.getByText('Cita de ejemplo');
      const authorElement = screen.getByText('Autor de ejemplo');
      expect(quoteElement).toBeInTheDocument();
      expect(authorElement).toBeInTheDocument();
    });
  });

  test('matches the snapshot', () => {
    const componentTree = renderer.create(<Quotes />);
    expect(componentTree.toJSON()).toMatchSnapshot();
  });

  test('returns updated object with AC', () => {
    const res = calculate({ total: '8', next: '9', operation: '-' }, 'AC');
    expect(res).toEqual({ total: null, next: null, operation: null });
  });

  test('update object with decimal', () => {
    const res = calculate({ total: '8', next: null, operation: null }, '.');
    expect(res).toEqual({ total: '8', next: '8.', operation: null });
  });

  test('simple operation with =', () => {
    const res = calculate({ total: '8', next: '2', operation: 'x' }, '=');
    expect(res).toEqual({ total: '16', next: null, operation: null });
  });

  test('simple operation with =', () => {
    const res = calculate({ total: '8', next: '2', operation: 'รท' }, '=');
    expect(res).toEqual({ total: '4', next: null, operation: null });
  });

  test('change sign', () => {
    const res = calculate({ total: '8', next: null, operation: null }, '+/-');
    expect(res).toEqual({ total: '-8', next: null, operation: null });
  });

  test('change sign next', () => {
    const res = calculate({ total: '8', next: '3', operation: '+' }, '+/-');
    expect(res).toEqual({ total: '8', next: '-3', operation: '+' });
  });

  test('should add two numbers correctly', () => {
    const res = operate(9, 3, '+');
    expect(res).toEqual('12');
  });
});

describe('operate', () => {
  test('should substract two numbers correctly', () => {
    const res = operate(9, 3, '-');
    expect(res).toEqual('6');
  });
});

describe('operate', () => {
  test('should multiply two numbers correctly', () => {
    const res = operate(9, 3, 'x');
    expect(res).toEqual('27');
  });
});

describe('operate', () => {
  test('should divide two numbers correctly', () => {
    const res = operate(9, 3, 'รท');
    expect(res).toEqual('3');
  });
});

describe('operate', () => {
  test('should calculate modulus of the number correctly', () => {
    const res = operate(9, 3, '%');
    expect(res).toEqual('0');
  });
});
