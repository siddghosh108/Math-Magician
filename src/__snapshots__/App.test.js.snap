// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests math-magician App test matches the snapshot 1`] = `<div />`;

exports[`tests math-magician App test renders the Items component 1`] = `
<button
  className="input-btn"
  onClick={[MockFunction]}
  type="button"
>
  8
</button>
`;

exports[`tests math-magician App test renders the Navbar component 1`] = `
<nav>
  <h1>
    Math Magicians
  </h1>
  <ul>
    <li>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        class=""
        href="/calculator"
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        class=""
        href="/quote"
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;

exports[`tests math-magician App test renders the QuotePage component 1`] = `
<div>
  <nav>
    <h1>
      Math Magicians
    </h1>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
  <main>
    <div />
  </main>
</div>
`;

exports[`tests math-magician App test should render the component 1`] = `
<div
  className="calc-container"
>
  <div
    className="output"
  >
    0
  </div>
  <button
    className="input-btn top"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="input-btn top"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="input-btn top"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="input-btn signs"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="input-btn"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="input-btn"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="input-btn"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="input-btn signs"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="input-btn"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="input-btn"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="input-btn"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="input-btn signs"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="input-btn"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="input-btn"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="input-btn"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="input-btn signs"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="input-btn zero"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="input-btn"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="input-btn signs"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;

exports[`tests math-magician App test should render the paragraph text 1`] = `
<p>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Elementum eu facilisis sed odio. Ullamcorper eget nulla facilisi etiam. Risus pretium quam vulputate dignissim suspendisse. Mi bibendum neque egestas congue quisque egestas diam in. Quis auctor elit sed vulputate mi sit amet mauris. Quam quisque id diam vel quam elementum. Vitae ultricies leo integer malesuada nunc. Nam at lectus urna duis. Integer feugiat scelerisque varius morbi. Tortor posuere ac ut consequat semper viverra nam libero. Pretium vulputate sapien nec sagittis aliquam. Varius vel pharetra vel turpis nunc eget lorem dolor sed. Lobortis scelerisque fermentum dui faucibus in ornare quam viverra orci.Fames ac turpis egestas sed. Fames ac turpis egestas sed tempus. Sed vulputate odio ut enim blandit. Vitae et leo duis ut diam quam nulla. Ullamcorper sit amet risus nullam eget felis. Velit ut tortor pretium viverra suspendisse. Egestas fringilla phasellus faucibus scelerisque eleifend donec pretium. Ornare arcu odio ut sem nulla pharetra diam sit. Volutpat sed cras ornare arcu dui vivamus arcu. Ultricies lacus sed turpis tincidunt id aliquet risus feugiat.
</p>
`;

exports[`tests math-magician App test should render the welcome message 1`] = `
<h2>
  Welcome to Math World!
</h2>
`;
